### ISSUE TITLE
`import plotly` crashes when `ipython` is installed for Python3.7

### ISSUE THREAD

**Issue by rmarren1** [plotly/plotly.py#1142]
It looks like this is really a bug in IPython.

When using Python3.7 with
```
plotly==0.2.7
ipython==0.6.5
```
`import plotly` breaks with the following traceback:

```
Python 3.7.0 (default, Aug  6 2018, 20:07:46) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import plotly
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/plotly/__init__.py", line 31, in <module>
    from plotly import (plotly, dashboard_objs, graph_objs, grid_objs, tools,
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/plotly/plotly/__init__.py", line 10, in <module>
    from . plotly import (
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/plotly/plotly/plotly.py", line 30, in <module>
    from plotly import exceptions, files, session, tools, utils
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/plotly/tools.py", line 59, in <module>
    ipython_core_display = optional_imports.get_module('IPython.core.display')
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/plotly/optional_imports.py", line 23, in get_module
    return import_module(name)
  File "/home/ryan/.virtualenvs/plotly/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/IPython/__init__.py", line 54, in <module>
    from .core.application import Application
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/IPython/core/application.py", line 25, in <module>
    from IPython.core import release, crashhandler
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/IPython/core/crashhandler.py", line 27, in <module>
    from IPython.core import ultratb
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/IPython/core/ultratb.py", line 120, in <module>
    from IPython.utils import path as util_path
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/IPython/utils/path.py", line 17, in <module>
    from IPython.utils.process import system
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/IPython/utils/process.py", line 19, in <module>
    from ._process_posix import system, getoutput, arg_split, check_pid
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/IPython/utils/_process_posix.py", line 23, in <module>
    import pexpect
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/pexpect/__init__.py", line 75, in <module>
    from .pty_spawn import spawn, spawnu
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/pexpect/pty_spawn.py", line 14, in <module>
    from .spawnbase import SpawnBase
  File "/home/ryan/.virtualenvs/plotly/local/lib/python3.7/site-packages/pexpect/spawnbase.py", line 224
    def expect(self, pattern, timeout=-1, searchwindowsize=-1, async=False):

```

This is because `async` is a reserved keyword in Python3.7+

**Comment by chriddyp** [comment]
Looks like it was fixed in the underlying `pexpect` package: https://github.com/pexpect/pexpect/issues/453

**Comment by jonmmease** [comment]
@rmarren1 what version of `pexpect` ended up installed in your environment?  According to pexpect/pexpect#453 this (or something similar) was fixed in `pexpect` version 4.3

**Comment by rmarren1** [comment]
Ah ok. Looks like some `requirements.txt` files in Dash install `pexpect<=4.3` causing the error. We should just update those.

Thanks!
